### PR TITLE
Add blocks listing

### PR DESCRIPTION
**Pull request by ahuang11**
@discdiver mentioned that it was hard to discover blocks in a Collection, so I decided to make a block listing.

I found a way to automate this. Once this gets approved, I can make one for the prefect-collection-template to propagate across our collections.

### Example

Automatically generated by gen_blocks_list.py

image

### Checklist

- [ ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-gcp/issues/new/choose) first.
- [ ] Includes tests or only affects documentation.
- [ ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-gcp/blob/main/CHANGELOG.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,14 +54,12 @@ To use Secret Manager:
...
-
+To use Vertex AI:
 ```bash
-prefect block register -m prefect_gcp.credentials
-prefect block register -m prefect_gcp.cloud_run
+pip install "prefect-gcp[aiplatform]"
 ```
 
-Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://orion-docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://orion-docs.prefect.io/ui/blocks/).
+Then, register to [view the blocks](https://orion-docs.prefect.io/ui/blocks/) on Prefect Cloud. A listing of available blocks in this collection can be found [here](https://prefecthq.github.io/prefect-gcp/blocks_listing/).
```
Comment (ahuang11) [inline README.md:62]: This link doesn't exist yet until this is merged.

Comment (discdiver) [reply README.md:62]: Thank you for the quick turn around! Is just naming it Blocks better? If not that, maybe Blocks List?

**Comment by ahuang11** [comment]
Here's what the gen_blocks_listing.py generated
image

File: README.md (hunk 1 above)
Comment (ahuang11) [reply README.md:62]: I like Blocks List. Blocks alone seems misleading.

File: mkdocs.yml [hunk 2]
```diff
@@ -50,6 +51,7 @@ watch:
 
 nav:
     - Home: index.md
+    - Blocks List: blocks_list.md
```
Comment (desertaxle) [inline mkdocs.yml:54]: Can we nest this under home so it isn't at the top level? Also, I have a weak opinion that we title the page `Blocks Catalog`, but I leave the decision up to you @ahuang11 
```suggestion
    - Home: index.md
        - Blocks Catalog: blocks_list.md
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 2 above)
Comment (ahuang11) [reply mkdocs.yml:54]: Seems a bit strange to me. Maybe I can append it to the README script generator.

image

Comment (ahuang11) [reply mkdocs.yml]: Okay, I inserted it under getting started:
image

Comment (ahuang11) [reply mkdocs.yml]: Also, nested the flows into their own section
image

**Review by desertaxle — APPROVED** [review]

**Pull request merged**